examples/go/pagerduty-trigger/handler/handler.go

```diff
@@ -0,0 +1,190 @@
+			http.StatusInternalServerError,
+			err,
+		)
+	}
+
+	// Display PagerDuty response.
+	return handler.Response{
+		Body:       pdRes,
+		StatusCode: http.StatusOK,
+	}, nil
+}
+
+func handlerResponseWithError(msg string, code int, err error) (handler.Response, error) {
+	wrapErr := fmt.Errorf(msg, err)
+	log.Println(wrapErr.Error())
+
+	return handler.Response{
+		Body:       []byte(wrapErr.Error()),
+		StatusCode: code,
+	}, wrapErr
+}
+
+func loadConfig(path string) (pdConfig, error) {
+	var pdc pdConfig
+
+	secret, err := toml.LoadFile(path)
+	if err != nil {
+		return pdConfig{}, fmt.Errorf("read pdconfig.toml: %w", err)
+	}
+
+	if err := secret.Unmarshal(&pdc); err != nil {
+		return pdConfig{}, fmt.Errorf("unmarshal pdconfig.toml: %w", err)
+	}
+
+	if err := validatePdConf(pdc); err != nil {
+		return pdConfig{}, err
+	}
+
+	return pdc, nil
+}
+
+func validatePdConf(pdc pdConfig) error {
+	if pdc.PagerDuty.RoutingKey == "" {
+		return errors.New("PagerDuty routing key cannot be empty in pdconfig.toml")
+	}
+
+	if pdc.PagerDuty.EventAction == "" {
+		return errors.New("PagerDuty event action cannot be empty in pdconfig.toml")
+	}
+
+	return nil
+}
+
+// parseEvent saves info from the triggering event into a PagerDuty data object.
+func parseEvent(req []byte, pdc pdConfig) (pdPayload, error) {
+	var ce cloudEvent
+	var pd pdPayload
+
+	if err := json.Unmarshal(req, &ce); err != nil {
+		return pd, err
+	}
+
+	if err := validateEvent(ce); err != nil {
+		return pd, err
+	}
+
+	pd.RoutingKey = pdc.PagerDuty.RoutingKey
+	pd.EventAction = pdc.PagerDuty.EventAction
+	pd.Client = "VMware Event Broker Appliance"
+	pd.ClientURL = ce.Source
+	pd.Payload.Summary = ce.Data.FullFormattedMessage
+	pd.Payload.Timestamp = ce.Data.CreatedTime
+	pd.Payload.Source = ce.Source
+	pd.Payload.Severity = "info"
+	pd.Payload.Component = ce.Data.Vm.Name
+	pd.Payload.Group = ce.Data.Host.Name
+	pd.Payload.Class = ce.Subject
+	pd.Payload.CustomDetails.User = ce.Data.UserName
+	pd.Payload.CustomDetails.Datacenter = ce.Data.Datacenter
+	pd.Payload.CustomDetails.ComputeResource = ce.Data.ComputeResource
+	pd.Payload.CustomDetails.Host = ce.Data.Host
+	pd.Payload.CustomDetails.VM = ce.Data.Vm
+
+	return pd, nil
+}
+
+func validateEvent(event cloudEvent) error {
+	var msg string
+
+	switch true {
+	case event.Source == "":
+		msg = "does not contain Source"
+	case event.Subject == "":
+		msg = "does not contain Subject"
+	case event.Data.FullFormattedMessage == "":
+		msg = "does not contain Data.FullFormattedMessage"
+	case (event.Data.CreatedTime == time.Time{}):
+		msg = "does not contain Data.CreatedTime"
+	case event.Data.Vm.Name == "":
+		msg = "does not contain Data.Vm.Name"
+	case event.Data.Host.Name == "":
+		msg = "does not contain Data.Vm.Name"
+	}
+
+	if msg != "" {
+		return errors.New("invalid event: " + msg)
+	}
+
+	return nil
+}
+
+func pdSendRequest(ctx context.Context, path string, pdp pdPayload) ([]byte, error) {
+	clt := &http.Client{}
```

Comment:
It's best practice to create the client with some same timeouts.

Reply:
Definite yes to that suggestion. I'll add some timeouts.